### PR TITLE
Implement WASI error-handling

### PR DESCRIPTION
**Pull request by kulcsaradam**
https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md#-errno-variant

Implement simple error-handling according to WASI documentation where functions return an error code as their first result and after that their usual results.

### CODE REVIEW

File: src/wasi/Wasi.cpp [hunk 1]
```diff
@@ -16,8 +16,12 @@
 
 #include "wasi/Wasi.h"
 
+// https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md
+
 namespace Walrus {
 
+std::string WASI::m_wasiErrorMessage[WASI::wasi_errno::notcapable + 1];
```
Comment (clover2123) [inline src/wasi/Wasi.cpp:23]: It seems that this string list is not used at all.
Would it be used later?

File: src/wasi/Wasi.cpp [hunk 2]
```diff
@@ -31,17 +35,20 @@ WASI::WasiFunc* WASI::find(std::string funcName)
 void WASI::test(ExecutionState& state, Value* argv, Value* result, void* data)
 {
     printf("No argument test succesful.\n");
+    result[0] = Value(static_cast<uint16_t>(WASI::wasi_errno::success));
 }
 
 void WASI::printI32(ExecutionState& state, Value* argv, Value* result, void* data)
 {
     printf("Recieved number: %d.\n", argv[0].asI32());
+    result[0] = Value(static_cast<uint16_t>(WASI::wasi_errno::success));
```
Comment (clover2123) [inline src/wasi/Wasi.cpp:44]: I cannot find error handling from the document you mentioned (https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/docs.md)
To be specific, I found `errno: Variant` but could find `test, printI32, and writeI32` functions.
Are these functions standard WASI features?

Comment (kulcsaradam) [reply src/wasi/Wasi.cpp:44]: No, these were just used for testing to see if arguments were handled right. They could be removed without any issues.

File: src/wasi/Wasi.cpp (hunk 1 above)
Comment (kulcsaradam) [reply src/wasi/Wasi.cpp:23]: I would like to use it to somehow communicate errors to the users of these functions, but I was unsure how you would like this part of wasi, so for the time being I left it it. It contains all the error messages in pairs with the error codes.

Comment (clover2123) [reply src/wasi/Wasi.cpp:23]: `walrus` aims to be a lightweight WASM engine for resource-limited devices.
So, allocating a global array just for error messages does not fit in our goal.
If you really need this array, please use it with debug mode only.

File: src/wasi/Wasi.cpp (hunk 2 above)
Comment (clover2123) [reply src/wasi/Wasi.cpp:44]: Please remove all non-standard WASI functions.
I think it's better to implement and test standard features only.

Comment (kulcsaradam) [reply src/wasi/Wasi.cpp]: I have removed the non-standard functions.

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**